test/common/3scale_crudl.go

```diff
@@ -69,3 +75,51 @@ func Test3ScaleCrudlPermissions(t TestingTB, ctx *TestingContext) {
 		t.Fatal(err)
 	}
 }
+
+func waitForUserToBecome3ScaleAdmin(t TestingTB, ctx *TestingContext, host, userName string) {
+	err := wait.PollImmediate(time.Second*10, time.Minute*5, func() (done bool, err error) {
+		systemSeedSecret := &corev1.Secret{}
+
+		err = ctx.Client.Get(goctx.TODO(), types.NamespacedName{Name: "system-seed", Namespace: ThreeScaleProductNamespace}, systemSeedSecret)
+		if err != nil {
+			t.Logf("unable to get system seed secret")
+			return false, nil
+		}
+
+		adminAccessToken := string(systemSeedSecret.Data["ADMIN_ACCESS_TOKEN"])
+
+		resp, err := ctx.HttpClient.Get(fmt.Sprintf("%s/admin/api/users.json?access_token=%s", host, adminAccessToken))
+		if err != nil {
+			t.Logf("unable to get list of users via api: %s", err)
+			return false, nil
+		}
+		defer resp.Body.Close()
```

Comment:
I was looking into this as I did not realize that we need to close the resp.Body and found this
```
When you call http.Get, the function returns a response as soon as all the HTTP headers have been read. The body of the response has not been read yet. The Response.Body is a wrapper around the network connection to the server. When you read from it, it downloads the body of the response.
```
Thanks Kevin!